packages/table/src/TableHead.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import HeaderRow from './HeaderRow';
+import TableHeader from './TableHeader';
+import { TableProps } from './Table';
+import { isComponentType } from '@leafygreen-ui/lib';
+import { useTableContext } from './TableContext';
+
+type TableHeaderProps<Shape> = Pick<TableProps<Shape>, 'columns'>;
+
+function TableHead<Shape>({ columns = [] }: TableHeaderProps<Shape>) {
+  const usingHeaderRow = React.useRef(false);
+
+  // Breaks tables with a colSpan when we don't subscribe to state here, trying to figure out why
+  // @ts-expect-error
+  const { state } = useTableContext(); // eslint-disable-line @typescript-eslint/no-unused-vars
```

Comment:
I don't feel comfortable merging this in if we don't know why. This isn't going to go away any time soon, if anything table will get more complex before it gets simpler. We should solve this when it's simpler.

Reply:
Found out what was going on, and solved for it :)